deltacat/tests/compute/compact_partition_test_cases.py

```diff
@@ -0,0 +1,450 @@
+import pyarrow as pa
+from typing import Callable, Dict, List
+from deltacat.tests.compute.test_util_common import (
+    offer_iso8601_timestamp_list,
+)
+from deltacat.tests.compute.test_util_constant import (
+    DEFAULT_MAX_RECORDS_PER_FILE,
+)
+from dataclasses import dataclass, fields
+
+from deltacat.compute.compactor.compaction_session import (
+    compact_partition_from_request as compact_partition_v1,
+)
+from deltacat.tests.compute.test_util_create_table_strategy import (
+    create_src_w_deltas_destination_strategy,
+    create_src_w_deltas_destination_rebase_w_deltas_strategy,
+)
+
+from deltacat.storage import (
+    DeltaType,
+)
+
+ENABLED_COMPACT_PARTITIONS_DRIVERS: List[Callable] = [compact_partition_v1]
+
+
+def create_tests_cases_for_all_compactor_versions(test_cases: Dict[str, List] = None):
+    if test_cases is None:
+        test_cases = {}
+    final_cases = {}
+    for version, compact_partition_func in enumerate(
+        ENABLED_COMPACT_PARTITIONS_DRIVERS
+    ):
+        for case_name, case_value in test_cases.items():
+            final_cases[f"{case_name}_v{version + 1}"] = [
+                *case_value,
+                compact_partition_func,
+            ]
+
+    return final_cases
+
+
+@dataclass(frozen=True)
+class CompactorTestCase:
+    primary_keys_param: Dict[str, str]
+    sort_keys_param: Dict[str, str]
+    partition_keys_param: List[Dict[str, str]]
+    partition_values_param: str
+    column_names_param: List[str]
+    input_deltas_arrow_arrays_param: List[pa.Array]
+    input_deltas_delta_type: str
+    expected_terminal_compact_partition_result: pa.Table
+    create_placement_group_param: bool
+    records_per_compacted_file_param: int
+    hash_bucket_count_param: int
+    validation_callback_func: Callable
```

Comment:
What is `validation_callback_func` used for? Is it used in any test?

Reply:
It was added in #188 for testing extensibility. The test case writer to define additional validations beyond just comparing the expected table values and actual table values after compaction. However, it seems like we haven't had a need for it so it may make sense to remove this test input parameter

Reply:
Removed